LevelUp.Pos.ProposedOrders/ProposedOrderCalculator.cs

```diff
@@ -40,139 +40,166 @@ public override string ToString()
+        /// <param name="exemptAmount">The current total of exempted items on the check, in cents.</param>
+        /// <param name="paymentAmount">The amount the customer would like to spend, in cents.</param>
         /// <param name="appliedDiscountAmount">The discount amount applied to the point of sale for the customer.</param>
         /// <returns>LevelUp.Pos.ProposedOrderCalculator.CalculateCompleteOrderValues</returns>
         public static AdjustedCheckValues CalculateCompleteOrderValues(
-            int totalOutstandingAmount,
-            int totalTaxAmount,
-            int totalExemptionAmount,
-            int customerPaymentAmount,
+            int outstandingAmount,
+            int taxAmount,
+            int exemptAmount,
+            int paymentAmount,
             int appliedDiscountAmount
         )
         {
-            int totalOutstandingAmountWithDiscount = totalOutstandingAmount + Math.Abs(appliedDiscountAmount);
+            int outstandingAmountWithDiscount = outstandingAmount + Math.Abs(appliedDiscountAmount);
 
-            int adjustedSpendAmount = CalculateAdjustedCustomerPaymentAmount(
-                customerPaymentAmount, 
-                totalOutstandingAmountWithDiscount);
+            return CalculateOrderValues(
+                outstandingAmountWithDiscount,
+                taxAmount,
+                exemptAmount,
+                paymentAmount);
+        }
 
-            int adjustedTaxAmount = CalculateAdjustedTaxAmount(
-                totalOutstandingAmountWithDiscount, 
-                totalTaxAmount,
-                adjustedSpendAmount);
+        internal static AdjustedCheckValues CalculateOrderValues(
+            int outstandingAmount,
+            int taxAmount,
+            int exemptionAmount,
+            int paymentAmount)
+        {
+            CheckData checkData = SanitizeData(outstandingAmount, taxAmount, exemptionAmount, paymentAmount);
+
+            var adjustedTaxAmount = CalculateAdjustedTaxAmount(checkData);
 
-            int adjustedExemptionAmount = CalculateAdjustedExemptionAmount(
-                totalOutstandingAmountWithDiscount,
-                totalTaxAmount,
-                totalExemptionAmount, 
-                adjustedSpendAmount);
+            var adjustedExemptionAmount = CalculateAdjustedExemptionAmount(checkData);
 
-            return new AdjustedCheckValues(adjustedSpendAmount, adjustedTaxAmount, adjustedExemptionAmount);
+            return new AdjustedCheckValues(checkData.PaymentAmount, adjustedTaxAmount, adjustedExemptionAmount);
         }
 
-        /// <summary>
-        /// A customer can not pay more than the largest amount due on a check.
-        /// </summary>
-        internal static int CalculateAdjustedCustomerPaymentAmount(int totalOutstandingAmount, int customerPaymentAmount)
+        internal static CheckData SanitizeData(int outstandingAmount, int taxAmount, int exemptionAmount, int paymentAmount)
```

Comment:
I like this sanitation encapsulation.  It seems intuitive to apply all of these sanity checks to how we process the check before applying our interpretation of the LevelUp platform business rules.